internal/server/server.go

```diff
@@ -167,20 +184,29 @@ func (s *server) CheckAccess(ctx context.Context, req *authorization.CheckAccess
 	// status. Otherwise, we return a denial if the result was an explicit denial.
 	switch {
 	case err == nil:
+		span.AddEvent("permitted")
+
 		out := &authorization.CheckAccessResponse{
 			Result: authorization.CheckAccessResponse_RESULT_ALLOWED,
 		}
 
 		return out, nil
 	case errors.Is(err, permissions.ErrUnauthenticated):
+		span.RecordError(err)
+
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	case errors.Is(err, permissions.ErrPermissionDenied):
+		span.RecordError(err)
```

Comment:
This isn't really an error, so we should just add an event:
```suggestion
		span.AddEvent("denied")
```